pyomo/core/expr/logical_expr.py

```diff
@@ -127,6 +87,16 @@ def __getstate__(self):
         return state
 
     def __nonzero__(self):
+        if not self.is_constant():
+            raise PyomoException('Cannot convert non-constant expression '
+                                 'to bool. This error usually arises for '
+                                 'using an expression in a boolean context '
+                                 'such as an if statement. For example, \n'
+                                 '    m.x = Var()\n'
+                                 '    e = inequality(-1, m.x, 1)'
+                                 '    if e:\n'
+                                 '        ...\n'
+                                 'would cause this exception.')
         return bool(self())
 
     __bool__ = __nonzero__
```

Comment:
Given @jsiirola's previous review comment about `__nonzero__` being a Python 2 thing, is this still needed?

Reply:
@blnicho Good catch! Thanks!